### PR TITLE
drush site-install fails with MySQL 8 #4259

### PR DESCRIPTION
**Pull request by AdamPS**
Drush 8 version of fix for #4259 that has been tested

### CODE REVIEW

**Comment by weitzman** [comment]
LGTM. Lets see what @greg-1-anderson  says as he is maintaining Drush8

**Comment by greg-1-anderson** [comment]
Did not see this earlier. Basically looks good to me; let's let it live on the 8.x branch for a while and see how it does.

I'll look into fixing the Drupal 9 failures separately; they appear unrelated.

**Comment by AdamPS** [comment]
Great thanks

File: lib/Drush/Sql/Sqlmysql.php [hunk 1]
```diff
@@ -93,8 +93,10 @@ public function createdb_sql($dbname, $quoted = FALSE) {
       // - If the database is on a remote server, create a wilcard user with %.
       //   We can't easily know what IP adderss or hostname would represent our server.
       $domain = ($this->db_spec['host'] == 'localhost') ? 'localhost' : '%';
-      $sql[] = sprintf('GRANT ALL PRIVILEGES ON %s.* TO \'%s\'@\'%s\'', $dbname, $this->db_spec['username'], $domain);
-      $sql[] = sprintf("IDENTIFIED BY '%s';", $this->db_spec['password']);
+      $user = sprintf("'%s'@'%s'", $this->db_spec['username'], $domain);
+      $sql[] = sprintf("DROP USER IF EXISTS %s;", $user);
```
Comment (totten) [inline lib/Drush/Sql/Sqlmysql.php:97]: Just a small aside (to save you time - if anyone else brings it up)... and maybe it doesn't matter if you don't need compat with older `mysqld`s... but this usage of `IF EXISTS` first appeared in MySQL 5.7 (compare: [5.6](https://dev.mysql.com/doc/refman/5.6/en/drop-user.html), [5.7](https://dev.mysql.com/doc/refman/5.7/en/drop-user.html)). So this probably doesn't work on MySQL 5.6 or earlier.

In another context (not specifically drush or drupal), I had a round of discussion with @seamuslee001 about this - and we couldn't find a single set of SQL statements that worked across versions. If it arises that you do need older `mysqld`s, then you may need more conditionals ([example](https://github.com/amp-cli/amp/pull/57/files)).

**Pull request merged**